packages/zk/src/proof/protocols/RLNProof.ts

```diff
@@ -1,39 +1,56 @@
-import { RLN, RLNFullProof } from "rlnjs";
+import { RLNProver, RLNSNARKProof } from "@cryptkeeperzk/rln-proof";
 
 import { ZkIdentitySemaphore } from "@src/identity";
 
-import type { RLNProofRequest } from "@cryptkeeperzk/types";
+import type { IRlnProofRequest, IRlnProverInputs } from "@cryptkeeperzk/types";
 
 import { IZkProof } from "./types";
-import { getMerkleProof, getRlnVerficationKeyJson } from "./utils";
+import { getMerkleProof, getMessageHash } from "./utils";
 
-export class RLNProofService implements IZkProof<RLNProofRequest, RLNFullProof> {
+export class RLNProofService implements IZkProof<IRlnProofRequest, RLNSNARKProof> {
   async genProof(
     identity: ZkIdentitySemaphore,
     {
+      rlnIdentifier,
+      message,
+      messageId,
+      messageLimit,
+      epoch,
       circuitFilePath,
       zkeyFilePath,
-      verificationKey,
-      merkleStorageAddress,
-      externalNullifier,
-      signal,
-      merkleProofArtifacts,
-      merkleProof: providerMerkleProof,
-    }: RLNProofRequest,
-  ): Promise<RLNFullProof> {
-    const rlnVerificationKeyJson = await getRlnVerficationKeyJson(verificationKey);
-
-    const rln = new RLN(circuitFilePath, zkeyFilePath, rlnVerificationKeyJson);
-
-    const identityCommitment = identity.genIdentityCommitment();
-
-    const merkleProof = await getMerkleProof({
-      identityCommitment,
       merkleProofArtifacts,
       merkleStorageAddress,
-      providerMerkleProof,
-    });
-
-    return rln.generateProof(signal, merkleProof, externalNullifier);
+      merkleProofProvided,
+    }: IRlnProofRequest,
+  ): Promise<RLNSNARKProof> {
+    const prover = new RLNProver(circuitFilePath, zkeyFilePath);
+
+    const rlnIdentifierBigInt = BigInt(rlnIdentifier);
+    const userMessageLimit = BigInt(messageLimit);
+    const messageHash = getMessageHash(message);
+    const identityCommitment = identity.genIdentityCommitment();
+    const identitySecret = identity.zkIdentity.getSecret();
+    const epochBigInt = typeof epoch === "bigint" ? epoch : BigInt(epoch);
```

Comment:
Does epoch have 'BigInt | string | number' type?

Reply:
Maybe we need to get rid of typeof check here if it's always a string.

Reply:
Makes sense, I will also get rid of other bigint ones.